merino-cache/src/memory.rs

```diff
@@ -379,15 +382,24 @@ mod tests {
         assert!(!cache.contains_key(&"expired".to_owned()));
 
         // Verify the reported metric.
-        assert_eq!(rx.len(), 2);
+        assert_eq!(rx.len(), 3);
         let collected_data: Vec<String> = rx
-            .iter()
-            .take(2)
+            .try_iter()
             .map(|x| String::from_utf8(x).unwrap())
-            .collect();
+            .collect::<Vec<String>>();
```

Comment:
Nit: `::<Vec<string>>` is not necessary here as it's already specified above by `collected_data`.

Reply:
Yes, I recall we discussed it @ncloudioj , I had it there to be extra explicit, but I understand it's definitely not required and redundant.